src/pyEQL/pourbaix/pourbaix_diagram.py

```diff
@@ -518,12 +518,15 @@ def __init__(
                 solid_entries = list(set(solid_pd.stable_entries) - set(entries_HO))
 
             self._filtered_entries = solid_entries + ion_entries
-            if len(comp_dict) > 1:
-                self._multi_element = True
-                self._processed_entries = self._preprocess_pourbaix_entries(self._filtered_entries, nproc=nproc)
-            else:
-                self._processed_entries = self._filtered_entries
-                self._multi_element = False
+
+            self._processed_entries = []
+            for sub_comp_dict in comp_dict:
+                if len(sub_comp_dict) > 1:
+                    self._multi_element = True
+                    processed_entries = self._preprocess_pourbaix_entries(self._filtered_entries, nproc=nproc)
```

Comment:
This line should be `self._processed_entries.extend(...)` rather than `processed_entries`. That's why it's failing I think